src/Components/test/testassets/ComponentsApp.App/Pages/Error.razor

```diff
@@ -0,0 +1,19 @@
+@page "/error"
+@using Microsoft.AspNetCore.Components
+<h1>Error</h1>
+@if(ShouldCauseError)
+{
+    throw new InvalidOperationException("Exception while rendering");    
+}
+else
+{
+    <button id="cause-error" onclick="@CauseError">Cause error</button>
+}
+@functions {
+    public bool ShouldCauseError { get; set; }
+
+    void CauseError()
+    {
+        ShouldCauseError = true;
+    }
+}
```

Comment:
This is really neat. I like how clear and simple it is.

One bit I'm not sure about is how we're distributing test code between `BasicTestApp` and `ComponentsApp.App`. Previously, `ComponentsApp.App` was meant to reflect more or less the default template state (but wired up with project references instead of package references), so we could quickly test out scenarios during development. It wasn't really intended as a place to store test cases.

Do you have any objection if I file a follow-up issue to normalize this a bit and migrate the new test cases over to `BasicTestApp` and restore `ComponentsApp.App/.Server` as reflecting an out-of-box normal app?

Reply:
I know `ComponentsApp.App` is within the "testassets" directory, which is not something I really think matches the goals. We just got told to put it there during the mondorepo move because that's apparently how other projects do it, and there wasn't a standard place to put "sandbox" type projects.

Reply:
That sounds good. In general I feel its more straightforward to add and try tests here. Maybe having a README.md on the E2E test project will help with making sure we add the tests in the right place.

Reply:
We should also have more completed tests covering exceptions. 
We should trigger exceptions from different points of the rendering pipeline and ensure they all do that this one does.

I'm going to go through the PR comments and file bugs for the things left here.